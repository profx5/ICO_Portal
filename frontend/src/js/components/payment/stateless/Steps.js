import React from 'react'
import styled from 'styled-components';
import {media} from 'js/utils/media';


const Steps = ({step, className}) => {

    return (
        <Wrapper className={className}>
            <Step step={step} passed={1 <= step} to_next={2<=step}>
                <div className="Step__head">Step 1</div>
                <div className="Step__desc">Verification (KYC)</div>
                <div className="Step__pointerBg"></div>
            </Step>
            <Step step={step} passed={2 <= step} to_next={3<=step}>
                <div className="Step__head">Step 2</div>
                <div className="Step__desc">Select payment method</div>
                <div className="Step__pointerBg"></div>
            </Step>
            <Step step={step} passed={3 <= step} to_next={4<=step}>
                <div className="Step__head">Step 3</div>
                <div className="Step__desc">Buying tokens</div>
                <div className="Step__pointerBg"></div>
            </Step>
            <Step step={step} passed={4 <= step}>
                <div className="Step__head">Step 4</div>
                <div className="Step__desc">Transaction mining</div>
                <div className="Step__pointerBg"></div>
            </Step>
        </Wrapper>
    );
}


export default Steps;

const Wrapper = styled.div`
    display: flex;
    justify-content: flex-start;
    flex-flow: row nowrap;
    margin-top: 20px;
    margin-bottom: 45px;
    &.centered {
        justify-content: center;
    }
    ${media.xs} {
        margin-top: 15px;
        margin-bottom: 7px;
    }
    ${media.sm} {
        margin-top: 30px;
    }
`;

const Step = styled.div`
    width: 190px;
    position: relative;
    ${media.smMinus} {
        display: none;
        width: 100%;
    }
    &:nth-of-type(1) {
        ${media.smMinus} {
            display: ${props => props.step === 1 && 'block'};
        }
    }
    &:nth-of-type(2) {
        margin: 0 110px;
        ${media.smMinus} {
            margin: 0;
            display: ${props => props.step === 2 && 'block'};
        }
    }
    &:nth-of-type(3) {
        margin-right: 110px;
        ${media.smMinus} {
            margin: 0;
            display: ${props => props.step === 3 && 'block'};
        }
    }
    &:nth-of-type(4) {
        &:before {
            display: none;
            ${media.smMinus} {
                display: block;
            }
        }
        ${media.smMinus} {
            display: ${props => props.step === 4 && 'block'};
        }
    }
    .Step__head {
        color: ${props => props.passed ? '#3679fc' : 'rgba(31, 31, 31,.3)'};
        font-size: 20px;
        font-weight: 500;
        letter-spacing: 0.5px;
        text-align: center;
        margin-bottom: 5px;
    }
    .Step__desc {
        color: ${props => props.passed ? '#000000' : 'rgba(31, 31, 31,.3)'};
        letter-spacing: 0.4px;
        font-weight: 500;
        text-align: center;
        white-space: nowrap;
        padding-bottom: 25px;
        font-size: 16px;
    }
    .Step__pointerBg {
        width: 65px;
        height: 65px;
        border-radius: 50px;
        display: none;
        background: #F5F6FA;
        position: absolute;
        left: 50%;
        bottom: -57px;
        transform: translate(-50%, -50%);
        z-index: 1;
        ${media.smMinus} {
            display: block;
        }
    }
    &:after {
        content: '';
        display: block;
        width: 16px;
        height: 16px;
        background: ${props => props.passed ? '#1767f2': 'rgba(0, 0, 0, .3)'};
        border: 2px solid white;
        box-shadow: 0 0 0 12px ${props => props.passed ? 'rgba(23, 103, 242, .07)': 'rgba(0, 0, 0, .02)'};
        border-radius: 100%;
        margin: 0 auto;
        z-index: 2;
        position: relative;
    }
    &:before {
        content: '';
        display: block;
        width: 247px;
        height: 2px;
        background: ${props => props.to_next ? '#1767f2': 'lightgrey'};
        position: absolute;
        bottom: 6px;
        left: calc(50% + 27px);
        transform: translateX(0);
        ${media.smMinus} {
            width: calc(100vw - 192px);;
            left: 50%;
            transform: translateX(-50%);
        }
        ${media.xs} {
            width: 130%;
            background: #419BF9;
        }
    }
`;
