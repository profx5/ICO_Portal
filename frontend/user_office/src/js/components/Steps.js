import React from 'react'
import styled from 'styled-components';

class Steps extends React.Component {
    render(step) {

        return (
            <Wrapper>
                <Step state={step}>
                    <div className="Step__head">Step 1</div>
                    <div className="Step__desc">Select payment method</div>
                </Step>
                <Step state={step}>
                    <div className="Step__head">Step 2</div>
                    <div className="Step__desc">Buying tokens</div>
                </Step>
                <Step state={step}>
                    <div className="Step__head">Step 3</div>
                    <div className="Step__desc">Transaction mining</div>
                </Step>
            </Wrapper>
        );
    }
}



export default Steps;

const Wrapper = styled.div`
    display: flex;
    justify-content: center;
    flex-flow: row nowrap;
`;

const Step = styled.div`
    width: 190px
    &:nth-of-type(2) {
        margin: 0 250px;
    }
    .Step__head {
        color: #3679fc;
        font-size: 20px;
        font-weight: 500;
        letter-spacing: 0.5px;
        text-align: center;
    }
    .Step__desc {
        color: #000000;
        letter-spacing: 0.4px;
        font-weight: 500;
        text-align: center;
        white-space: nowrap;
        padding-bottom: 25px;
    }
    &:after {
        content: '';
        display: block;
        width: 16px;
        height: 16px;
        background: #1767f2;
        border: 2px solid white;
        box-shadow: 0 0 0 12px rgba(23,103,242,0.07);
        border-radius: 100%;
        margin: 0 auto;
    }
    &:not(:last-of-type) {
        position: relative;
        &:before {
            content: '';
            display: block;
            width: 387px;
            height: 2px;
            background: #1767f2;
            position: absolute;
            bottom: 6px;
            left: calc(50% + 27px);
            transform: translateX(0);
        }
    }
`;