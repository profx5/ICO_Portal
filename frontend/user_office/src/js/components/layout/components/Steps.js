import React from 'react';
import styled from 'styled-components';

import arrowImg from './../../../../img/arrow-down.svg';
import exclaimIcon from './../../../../img/icons/icon_exclaim.svg';
import checkIcon from './../../../../img/check_mini_icon.svg';
import userPlaceholder from './../../../../img/user.svg';


const Steps = ({stepsPassed, onClickHandler, isDropdownAccountOpen}) => {
    return (
        <Wrapper className="DropdownAccountTrigger" onClick={onClickHandler}>
            <StyledSteps>Steps completed: <span>{stepsPassed}/3</span></StyledSteps>
            <UserImg approved={stepsPassed === 3}/>
            <ArrowImg up={isDropdownAccountOpen} src={arrowImg}/>
        </Wrapper>
    )
}


export default Steps;

const Wrapper = styled.div`
    display: inline-flex;
    align-items: center;
    cursor: pointer;
    position: relative;
    z-index: 1;
    & * {
        pointer-events: none;
    }
`;

const StyledSteps = styled.p`
    color: #222121;
    user-select: none;
    span {
        color: #00da36;
        font-size: 16px;
        font-weight: 600;
    }
`;

const UserImg = styled.div`
    width: 47px;
    height: 47px;
    margin-left: 20px;
    position: relative;
    z-index: 1;
    background: url(${userPlaceholder}) no-repeat center;
    background-size: cover;
    &:before {
        content: '';
        width: 18px;
        height: 18px;
        border-radius: 100%;
        background: url(${props => props.approved ? checkIcon : exclaimIcon}) no-repeat center, #f26d6d;
        position: absolute;
        bottom: -4px;
        right: 0px;
        border: 2px solid white;
    }
`;

const ArrowImg = styled.img`
    margin-left: 20px;
    position: relative;
    top: 1px;
    transform: ${props => props.up ? 'rotate(0)' : 'rotate(-180deg)'};
`;
