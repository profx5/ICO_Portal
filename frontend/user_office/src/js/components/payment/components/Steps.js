import React from 'react'
import styled from 'styled-components';


const Steps = ({step}) => {

    return (
        <Wrapper>
            <Step passed={1 <= step} to_next={2<=step}>
                <div className="Step__head">Step 1</div>
                <div className="Step__desc">Select payment method</div>
            </Step>
            <Step passed={2 <= step} to_next={3<=step}>
                <div className="Step__head">Step 2</div>
                <div className="Step__desc">Buying tokens</div>
            </Step>
            <Step passed={3 <= step}>
                <div className="Step__head">Step 3</div>
                <div className="Step__desc">Transaction mining</div>
            </Step>
        </Wrapper>
    );
}


export default Steps;

const Wrapper = styled.div`
    display: flex;
    justify-content: center;
    flex-flow: row nowrap;
`;

const Step = styled.div`
    width: 190px
    &:nth-of-type(2) {
        margin: 0 250px;
    }
    .Step__head {
        color: #3679fc;
        font-size: 20px;
        font-weight: 500;
        letter-spacing: 0.5px;
        text-align: center;
        margin-bottom: 5px;
    }
    .Step__desc {
        color: #000000;
        letter-spacing: 0.4px;
        font-weight: 500;
        text-align: center;
        white-space: nowrap;
        padding-bottom: 25px;
        font-size: 16px;
    }
    &:after {
        content: '';
        display: block;
        width: 16px;
        height: 16px;
        background: ${props => props.passed ? '#1767f2': 'lightgrey'};
        border: 2px solid white;
        box-shadow: 0 0 0 12px rgba(23,103,242,0.07);
        border-radius: 100%;
        margin: 0 auto;
    }
    &:not(:last-of-type) {
        position: relative;
        &:before {
            content: '';
            display: block;
            width: 387px;
            height: 2px;
            background: ${props => props.to_next ? '#1767f2': 'lightgrey'};
            position: absolute;
            bottom: 6px;
            left: calc(50% + 27px);
            transform: translateX(0);
        }
    }
`;
