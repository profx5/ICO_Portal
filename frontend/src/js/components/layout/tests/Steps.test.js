import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Steps from 'js/components/layout/stateless/Steps';


configure({ adapter: new Adapter() });

it('should render stateless Steps', () => {
    const component = shallow(<Steps/>);
    expect(component).toMatchSnapshot();
})
