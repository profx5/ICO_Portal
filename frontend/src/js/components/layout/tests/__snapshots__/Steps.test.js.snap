// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render stateless Steps 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Steps />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef>
          Steps completed: 
          <span>
            /2
          </span>
        </ForwardRef>,
        <ForwardRef
          approved={false}
        />,
        <ForwardRef
          src="arrow-down.svg"
        />,
      ],
      "className": "DropdownAccountTrigger",
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            "Steps completed: ",
            <span>
              /2
            </span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Steps completed: ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                undefined,
                "/2",
              ],
            },
            "ref": null,
            "rendered": Array [
              undefined,
              "/2",
            ],
            "type": "span",
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
    color: #222121;
    user-select: none;
    white-space: nowrap;
    span {
        color: #00da36;
        font-size: 16px;
        font-weight: 600;
    }
    ",
              "@media (max-width: 767px)",
              " {
        display: none;
    }
",
            ],
          },
          "displayName": "styled.p",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "p",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "approved": false,
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": false,
            "rules": Array [
              "
    width: 47px;
    height: 47px;
    margin-left: 20px;
    position: relative;
    z-index: 1;
    background: url(",
              "user.svg",
              ") no-repeat center;
    background-size: cover;
    &:before {
        content: '';
        width: 18px;
        height: 18px;
        border-radius: 100%;
        background: url(",
              [Function],
              ") no-repeat center, #f26d6d;
        position: absolute;
        bottom: -4px;
        right: 0px;
        border: 2px solid white;
    }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "src": "arrow-down.svg",
          "up": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bxivhb",
            "isStatic": false,
            "rules": Array [
              "
    margin-left: 20px;
    position: relative;
    top: 1px;
    transform: ",
              [Function],
              ";
    ",
              "@media (max-width: 767px)",
              " {
        margin-left: 9px;
    }
",
            ],
          },
          "displayName": "styled.img",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bxivhb",
          "target": "img",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
    display: inline-flex;
    align-items: center;
    cursor: pointer;
    position: relative;
    z-index: 1;
    & * {
        pointer-events: none;
    }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef>
            Steps completed: 
            <span>
              /2
            </span>
          </ForwardRef>,
          <ForwardRef
            approved={false}
          />,
          <ForwardRef
            src="arrow-down.svg"
          />,
        ],
        "className": "DropdownAccountTrigger",
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              "Steps completed: ",
              <span>
                /2
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Steps completed: ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  undefined,
                  "/2",
                ],
              },
              "ref": null,
              "rendered": Array [
                undefined,
                "/2",
              ],
              "type": "span",
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "rules": Array [
                "
    color: #222121;
    user-select: none;
    white-space: nowrap;
    span {
        color: #00da36;
        font-size: 16px;
        font-weight: 600;
    }
    ",
                "@media (max-width: 767px)",
                " {
        display: none;
    }
",
              ],
            },
            "displayName": "styled.p",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "p",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "approved": false,
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": false,
              "rules": Array [
                "
    width: 47px;
    height: 47px;
    margin-left: 20px;
    position: relative;
    z-index: 1;
    background: url(",
                "user.svg",
                ") no-repeat center;
    background-size: cover;
    &:before {
        content: '';
        width: 18px;
        height: 18px;
        border-radius: 100%;
        background: url(",
                [Function],
                ") no-repeat center, #f26d6d;
        position: absolute;
        bottom: -4px;
        right: 0px;
        border: 2px solid white;
    }
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "src": "arrow-down.svg",
            "up": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bxivhb",
              "isStatic": false,
              "rules": Array [
                "
    margin-left: 20px;
    position: relative;
    top: 1px;
    transform: ",
                [Function],
                ";
    ",
                "@media (max-width: 767px)",
                " {
        margin-left: 9px;
    }
",
              ],
            },
            "displayName": "styled.img",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bxivhb",
            "target": "img",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
    display: inline-flex;
    align-items: center;
    cursor: pointer;
    position: relative;
    z-index: 1;
    & * {
        pointer-events: none;
    }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
